keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -872,6 +898,12 @@ enhancement:
   cluster required to make on upgrade, in order to make use of the enhancement?
 -->
 
+kube-apiserver, kubelet and CRI implementation have to support the feature for it to work e2e.
```

Comment:
I called it out earlier but what is meant by "CRI Implementation supports the feature"?

Will this be feature gated at the container runtime level? Or will it be a enabled for all right when it goes into containerd/CRIO?

How do we handle skew of container runtime against kubelet/kubeapi-server? Containerd will have many production users that may not have this feature enabled so it would be worth thinking through the UX in this case.

Reply:
> I called it out earlier but what is meant by "CRI Implementation supports the feature"?

It basically turns down to if the `Mount` structure uses the `Image` field as indicator (at least for CRI-O):
https://github.com/kubernetes/cri-api/blob/82e1e9174de30efdb4f82fbb5251c31e4a7ce34a/pkg/apis/runtime/v1/api.proto#L257

> Will this be feature gated at the container runtime level? Or will it be a enabled for all right when it goes into containerd/CRIO?

For CRI-O there will be no feature gate, I can't speak how containerd will handle it.

Reply:
> For CRI-O the volume will just not appear. I assume the only way to resolve this would be an additional runtime config field within the CRI.

Is that a runtime handler?

Reply:
@kannon92 no, I was more thinking about adding it to: https://github.com/kubernetes/cri-api/blob/82e1e9174de30efdb4f82fbb5251c31e4a7ce34a/pkg/apis/runtime/v1/api.proto#L135-L142

Reply:
This seems to be something that is important for containerd integration but that is not yet in scope for beta off. It is in scope for beta on though.